apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -1206,21 +1207,20 @@ impl FetchDependencyGraph {
             if condition.is_interface_type() {
                 // Lastly, we just need to check that we're coming from a subgraph
                 // that has the type as an interface object in its schema.
-                Ok(self.parents_of(node_index).any(|p| {
-                    let Ok(p_node) = self.node_weight(p) else {
-                        return false;
-                    };
-                    let p_subgraph_name = &p_node.subgraph_name;
-                    let Ok(p_subgraph_schema) = get_subgraph_schema(p_subgraph_name) else {
-                        return false;
-                    };
-                    let Ok(type_in_parent) =
-                        p_subgraph_schema.get_type(condition.type_name().clone())
-                    else {
-                        return false;
-                    };
-                    type_in_parent.is_interface_object_type(&p_subgraph_schema)
-                }))
+                Ok(self
+                    .parents_of(node_index)
+                    .map(|p| {
+                        let p_node = self.node_weight(p)?;
+                        let p_subgraph_name = &p_node.subgraph_name;
+                        let p_subgraph_schema = get_subgraph_schema(p_subgraph_name)?;
+                        let Ok(type_in_parent) =
+                            p_subgraph_schema.get_type(condition.type_name().clone())
+                        else {
+                            return Ok(false);
```

Comment:
I see the JS does the same, but how intentional is it that this returns `false` instead of an error? Can we add a comment if it is okay if the subgraph schema doesn't have that type?